### PR TITLE
Fix BUILD_STATIC

### PR DESCRIPTION
**Pull request by InspectorConstructor**
- Change BUILD_STATIC to BUILD_STATIC_LIBS
- ~Use target_link_options() to build the two libs statically instead of CMAKE_EXE_LINKER_FLAGS, which sets the -static flag globally~ Remove `-static` global linker flag

(By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.)

### CODE REVIEW

**Comment by InspectorConstructor** [comment]
After trying a few different things, I realized that using `-static` while linking the library is not needed, since we declare the libraries as either STATIC or SHARED in add_library().

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,9 +2,9 @@ cmake_minimum_required(VERSION 2.8)
 project(KinesisVideoWebRTCClient LANGUAGES C)
 
 # User Flags
-option(BUILD_STATIC "Build with static linkage." OFF)
```
Comment (sharadrj) [inline CMakeLists.txt]: Readme should be updated: https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#cmake-arguments

File: .gitmodules [hunk 2]
```diff
@@ -3,4 +3,4 @@
 	url = https://github.com/awslabs/amazon-kinesis-video-streams-pic.git
 [submodule "open-source/amazon-kinesis-video-streams-producer-c"]
 	path = open-source/amazon-kinesis-video-streams-producer-c
-	url = https://github.com/awslabs/amazon-kinesis-video-streams-producer-c.git
+	url = https://github.com/InspectorConstructor/amazon-kinesis-video-streams-producer-c.git
```
Comment (chehefen) [inline .gitmodules]: can you fix the submodule url? thanks

Comment (InspectorConstructor) [reply .gitmodules]: sorry! yes, i'm undoing that commit.

**Comment by chehefen** [comment]
Hi @InspectorConstructor, can you rebase your change on top of this commit: https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/commit/63d89964a2aa38c08deb9bda5bd36ca633266a7b, then the travis check should pass.

**Comment by InspectorConstructor** [comment]
I struggled with the rebase, so I did a merge instead. Is this acceptable? If you have the ability to "Squash and Merge" it should achieve the same thing.

**Comment by chehefen** [comment]
> I struggled with the rebase, so I did a merge instead. Is this acceptable? If you have the ability to "Squash and Merge" it should achieve the same thing.

Just approved. Can you squash your commits so that it looks cleaner? Thanks

**Pull request closed without merging**